src/converter/pytorch_converter.py

```diff
@@ -242,13 +244,15 @@ def convert_json_to_protobuf_nodes(
                             [
                                 ChakraAttr(name="comm_type", int64_val=collective_comm_type),
                                 ChakraAttr(name="comm_size", int64_val=pytorch_gpu_node.comm_size),
+                                ChakraAttr(name="pg_name", string_val=pytorch_gpu_node.pg_name),
```

Comment:
Do you assume that pg_name is always available in pytorch_gpu_node? It seems that your assumption about the field is not consistent. In the KinetoOperator class, it is an optional field, while the PyTorch node class considers it essential.

Reply:
i)  
No, I don't assume that pg_name is always available in pytorch_gpu_node. 
Line number: 241 insures that it is always `COMM_COLL_NODE` by "if chakra_gpu_node.type == COMM_COLL_NODE:".
ii) 
I just checked what happends if "pg_name" is none. pg name will be encoded as "".
```
   {
     "name": "pg_name",
     "stringVal": ""
   }
```

Reply:
I think it is better not to add the pg_name field when it is None. The current Chakra schema is not capacity-efficient because it repeats many fields. If we encode pg_name even if it is None, then we are wasting bytes. Let's add the field conditionally.

Reply:
I agree. Just update not to put pg_name field in the traces. 

Below is the result from ASTRA-Sim.
```
Node Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
No PG Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
Node Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
No PG Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
Node Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
No PG Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
Node Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
No PG Name: ncclDevKernel_AllGather_RING_LL(ncclDevKernelArgsStorage<4096ul>)
Node Name: ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)
PG name: 31
Node Name: ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)
PG name: 34
Node Name: ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)
PG name: 40
Node Name: ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)
PG name: 37
Node Name: ncclDevKernel_AllReduce_Sum_bf16_RING_LL(ncclDevKernelArgsStorage<4096ul>)
PG name: 28
Node Name: ncclDevKernel_AllReduce_Sum_bf16_RING_LL(ncclDevKernelArgsStorage<4096ul>)
PG name: 27
....
```